Source/WebCore/rendering/RenderBlock.cpp

```diff
@@ -1102,6 +1102,18 @@ void RenderBlock::markForPaginationRelayoutIfNeeded()
         setChildNeedsLayout(MarkOnlyThis);
 }
 
+LayoutRect RenderBlock::computeRectForRepaint(const LayoutRect& rect, const RenderLayerModelObject* repaintContainer) const
```

Comment:
Overriding `computeRectForRepaint` isn’t the right way to handle this.

First, let’s rename `repaintCaretRectForLocalRect()` which is confusing—it’s not doing a repaint, it’s computing a rectangle, so needs to be called `caretRepaintRectForLocalRect()`.

`repaintCaretForLocalRect()` needs the special-case “non-clipped repaint rect” logic. So you need to figure out where the repaint rect is getting clipped, and modify that behavior based on this special case.

Reply:
It gets clipped because `repaintCaretForLocalRect` in FrameSelection.cpp calls `RenderObject::repaintRectangle` which calls `RenderObject::issueRepaint` and the rect is clipped by this line in RenderObject.cpp by `computeRectForRepaint`:
```
auto repaintRect = partialRepaintRect ? computeRectForRepaint(*partialRepaintRect, repaintContainer.renderer) : clippedOverflowRectForRepaint(repaintContainer.renderer);
```

Specifically within `computeRectForRepaint` the clip occurs in `RenderBox::computeVisibleRectInContainer` during the call to `repaintCaretForLocalRect`


I could limit the impact to the the call to `repaintRectangle` in `repaintCaretForLocalRect`, that seems like a better idea.